deploy/8-deploy-sfrxeth-oracle.ts

```diff
@@ -0,0 +1,50 @@
+import { ethers } from "hardhat";
+import { DeployFunction } from "hardhat-deploy/dist/types";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { ADDRESSES } from "../helpers/deploymentConfig";
+
+const func: DeployFunction = async ({ getNamedAccounts, deployments, network }: HardhatRuntimeEnvironment) => {
+  const { deploy } = deployments;
+  const { deployer } = await getNamedAccounts();
+
+  const proxyOwnerAddress = network.live ? ADDRESSES[network.name].timelock : deployer;
+
+  const { sfrxETH, SfrxEthFraxOracle } = ADDRESSES[network.name];
+
+  let SfrxEthFraxOracleAddress = SfrxEthFraxOracle;
+  if (!SfrxEthFraxOracle) {
+    await deploy("MockSfrxEthFraxOracle", {
+      contract: "MockSfrxEthFraxOracle",
+      from: deployer,
+      log: true,
+      autoMine: true,
+      skipIfAlreadyDeployed: true,
+      args: [],
+    });
+
+    const mockSfrxEthFraxOracle = await ethers.getContract("MockSfrxEthFraxOracle");
+    SfrxEthFraxOracleAddress = mockSfrxEthFraxOracle.address;
+
+    if ((await mockSfrxEthFraxOracle.owner()) === deployer) {
+      await mockSfrxEthFraxOracle.transferOwnership(proxyOwnerAddress);
+    }
+  }
+
+  await deploy("SFrxETHOracle", {
+    contract: "SFrxETHOracle",
+    from: deployer,
+    log: true,
+    deterministicDeployment: false,
+    args: [SfrxEthFraxOracleAddress, sfrxETH],
+    proxy: {
+      owner: proxyOwnerAddress,
+      proxyContract: "OptimizedTransparentProxy",
+    },
+    skipIfAlreadyDeployed: true,
```

Comment:
Why don't we execute the `initialize` function here, as we made for the ResilientOracle, for example?